File: src/profed/models/resume.py
# Copyright (C) 2026 Christof Donat
# SPDX-License-Identifier: AGPL-3.0-or-later

from pydantic import BaseModel, ConfigDict, Field
from typing import Any


class Resume(BaseModel):
    model_config = ConfigDict(extra="allow")

    experience: list[dict[str, Any]] = Field(default_factory=list)
    education: list[dict[str, Any]] = Field(default_factory=list)
    skills: list[dict[str, Any]] = Field(default_factory=list)
    projects: list[dict[str, Any]] = Field(default_factory=list)

File: client/src/lib/resume.ts
// Copyright (C) 2026 Christof Donat
// SPDX-License-Identifier: AGPL-3.0-or-later

import type { Resume } from '$lib/api/types';

export function hasResumeContent(resume: Resume | null | undefined): boolean {
    if ( !resume ) {
        return false;
    }
    return (resume.experience.length > 0 ||
            resume.education.length > 0 ||
            resume.skills.length > 0 ||
            resume.projects.length > 0);
}

